src/controllers/nodes/oneagent_dao.go

```diff
@@ -2,34 +2,29 @@ package nodes
 
 import (
 	"context"
-	"os"
 
 	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (r *NodesController) determineDynakubeForNode(nodeName string) (*dynatracev1beta1.DynaKube, error) {
-	dkList, err := r.getOneAgentList()
+func (controller *NodesController) determineDynakubeForNode(nodeName string) (*dynatracev1beta1.DynaKube, error) {
+	dks, err := controller.getDynakubeList()
 	if err != nil {
 		return nil, err
 	}
-
-	return r.filterOneAgentFromList(dkList, nodeName), nil
+	return controller.filterOneAgentFromList(dks, nodeName), nil
 }
 
-func (r *NodesController) getOneAgentList() (*dynatracev1beta1.DynaKubeList, error) {
-	watchNamespace := os.Getenv("POD_NAMESPACE")
-
-	var dkList dynatracev1beta1.DynaKubeList
-	err := r.client.List(context.TODO(), &dkList, client.InNamespace(watchNamespace))
+func (controller *NodesController) getDynakubeList() (*dynatracev1beta1.DynaKubeList, error) {
+	var dynakubeList dynatracev1beta1.DynaKubeList
+	err := controller.client.List(context.TODO(), &dynakubeList, client.InNamespace(controller.podNamespace))
 	if err != nil {
 		return nil, err
 	}
-
-	return &dkList, nil
+	return &dynakubeList, nil
 }
 
-func (r *NodesController) filterOneAgentFromList(dkList *dynatracev1beta1.DynaKubeList,
+func (controller *NodesController) filterOneAgentFromList(dkList *dynatracev1beta1.DynaKubeList,
```

Comment:
`filterDynaKubeFromList`
you are filtering the dynakubes (using the oneagent instances)